src/core/public/chrome/ui/header/header_breadcrumbs.tsx

```diff
@@ -51,15 +51,27 @@ export function HeaderBreadcrumbs({ appTitle$, breadcrumbs$, breadcrumbsAppendEx
     ),
   }));
 
-  if (breadcrumbsAppendExtension) {
-    const lastCrumb = crumbs[crumbs.length - 1];
-    lastCrumb.text = (
-      <>
-        {lastCrumb.text}
-        <HeaderExtension extension={breadcrumbsAppendExtension.content} display={'inlineBlock'} />
-      </>
-    );
-  }
+  const Breadcrumbs = (
+    <EuiHeaderBreadcrumbs
+      breadcrumbs={crumbs}
+      max={10}
+      data-test-subj="breadcrumbs"
+      className={classNames({
+        'chrHeaderNavBreadcrumbs--withAppendExtension': !!breadcrumbsAppendExtension,
+      })}
+    />
+  );
+
+  if (!breadcrumbsAppendExtension) return Breadcrumbs;
 
-  return <EuiHeaderBreadcrumbs breadcrumbs={crumbs} max={10} data-test-subj="breadcrumbs" />;
+  return (
+    <EuiFlexGroup responsive={false} gutterSize={'none'} className="eui-textTruncate">
+      <EuiFlexItem grow={false} className="eui-textTruncate">
```

Comment:
Rather than using `className="eui-textTruncate"`, could we simply apply a style of `min-width: 0;` to both of these elements? It will achieve the same effect, but without applying extraneous styles.

Reply:
Not sure if this tip is still relevant for new approach, I left a single `className="eui-textTruncate"` and wasn't able to get rid of it

Reply:
Yeah, this no longer applies with your new approach.